src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LbfgsPredictorBase.cs

```diff
@@ -51,10 +51,10 @@ public abstract class OptionsBase : TrainerInputBaseWithWeight
             /// Number of previous iterations to remember for estimate of Hessian.
             /// </summary>
             [Argument(ArgumentType.AtMostOnce, HelpText = "Memory size for L-BFGS. Low=faster, less accurate",
-                ShortName = "m, MemorySize", SortOrder = 50)]
+                Name = "HistorySize", ShortName = "m, MemorySize", SortOrder = 50)]
             [TGUI(Description = "Memory size for L-BFGS", SuggestedSweeps = "5,20,50")]
             [TlcModule.SweepableDiscreteParamAttribute("MemorySize", new object[] { 5, 20, 50 })]
-            public int HistorySize = Defaults.HistorySize;
+            public int HistoryLength = Defaults.HistoryLength;
```

Comment:
>HistoryLength [](start = 23, length = 13)

[https://github.com/dotnet/machinelearning/issues/2882](https://github.com/dotnet/machinelearning/issues/2882)
I though we agreed on `HistorySize`
Why this sudden change?

Reply:
I didn't in that PR (sorry for missing the associated issue). `Size` usually means number of bytes to developers.

Reply:
`HistorySize` is standard nomenclature in the optimization field, so I'd prefer to keep it so. I don't think `Size` necessarily means size of memory (but I could be wrong :)).

---
In reply to: [267485468](https://github.com/dotnet/machinelearning/pull/3029#discussion_r267485468) [](ancestors = 267485468)

Reply:
How would users describe the `size` of history? Length? Size? I like `Length` a bit more but I am ok with the other choice too, so let's do `Size`.

Reply:
We already had that discussion and we come up with `HistorySize`.
This is gonna be third time we rename this parameter, and I'm against another renaming for sake of renaming.

---
In reply to: [267585933](https://github.com/dotnet/machinelearning/pull/3029#discussion_r267585933) [](ancestors = 267585933)

Reply:
Yes, so `Size` is used.

Reply:
Looks like I miss some `HistoryLength`s. I am fixing them.